File: BOJ/2448.cpp
#include<cstdio>
using namespace std;

int arr[3073][6145];

void star(int y, int x, int n) {
	if (n == 0) return;
	if (n % 3 == 0) {
		arr[y][x] = '*';
		arr[y + 1][x - 1] = '*';
		arr[y + 1][x + 1] = '*';
		arr[y + 2][x - 2] = '*';
		arr[y + 2][x - 1] = '*';
		arr[y + 2][x] = '*';
		arr[y + 2][x + 1] = '*';
		arr[y + 2][x + 2] = '*';
	}

	star(y, x, n / 2);
	star(y + n / 2, x - n / 2, n / 2);
	star(y + n / 2, x + n / 2, n / 2);
}

int main() {
	int n;
	scanf("%d", &n);
	for (int i = 0; i < n; i++) {
		for (int j = 0; j < 2 * n; j++) {
			arr[i][j] = ' ';
		}
	}

	star(0, n - 1, n);

	for (int i = 0; i < n; i++) {
		for (int j = 0; j < 2 * n - 1; j++) {
			printf("%c", arr[i][j]);
		}
		printf("\n");
	}

	return 0;
}

File: BOJ/python/2448.py
import sys

input = sys.stdin.readline

N = int(input().rstrip("\n"))

arr = [[" " for _ in range(N * 2)] for _ in range(N)]


def star(y, x, size):
    if size == 0:
        return
    if size % 3 == 0:
        arr[y][x] = "*"
        arr[y + 1][x - 1] = "*"
        arr[y + 1][x + 1] = "*"
        arr[y + 2][x - 2] = "*"
        arr[y + 2][x - 1] = "*"
        arr[y + 2][x] = "*"
        arr[y + 2][x + 1] = "*"
        arr[y + 2][x + 2] = "*"

    next_size = size // 2

    star(y, x, next_size)
    star(y + next_size, x - next_size, next_size)
    star(y + next_size, x + next_size, next_size)


star(0, N - 1, N)

for i in range(N):
    print("".join(arr[i]))
